File: test/golden/encrypt_parity.py
#!/usr/bin/env python3
"""Deterministic encryption parity helper for Go test harness."""

from __future__ import annotations

import base64
import binascii
import os
import sys
from pathlib import Path

ROOT = Path(__file__).resolve().parents[2]
sys.path.insert(0, str(ROOT / "sdk" / "python"))

from atb.encrypt import encrypt_raw  # noqa: E402


def _required(name: str) -> str:
    value = os.environ.get(name, "")
    if not value:
        raise RuntimeError(f"missing required env var {name}")
    return value


def main() -> int:
    plaintext = base64.b64decode(_required("ATB_PARITY_PLAINTEXT_B64"))
    password = _required("ATB_PARITY_PASSWORD")
    salt = bytes.fromhex(_required("ATB_PARITY_SALT_HEX"))
    nonce = bytes.fromhex(_required("ATB_PARITY_NONCE_HEX"))

    encrypted = encrypt_raw(plaintext, password, salt=salt, nonce=nonce)
    sys.stdout.write(binascii.hexlify(encrypted).decode("ascii"))
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: test/golden/encrypt_parity.js
#!/usr/bin/env node
"use strict";

const { encryptRaw } = require("../../sdk/typescript/dist/index.js");

function required(name) {
  const value = process.env[name];
  if (!value) {
    throw new Error(`missing required env var ${name}`);
  }
  return value;
}

async function main() {
  const plaintext = Buffer.from(
    required("ATB_PARITY_PLAINTEXT_B64"),
    "base64"
  );
  const password = required("ATB_PARITY_PASSWORD");
  const salt = Buffer.from(required("ATB_PARITY_SALT_HEX"), "hex");
  const nonce = Buffer.from(required("ATB_PARITY_NONCE_HEX"), "hex");

  const encrypted = encryptRaw(plaintext, password, { salt, nonce });
  process.stdout.write(Buffer.from(encrypted).toString("hex"));
}

main().catch((err) => {
  console.error(err && err.stack ? err.stack : String(err));
  process.exit(1);
});
